### PR TITLE
Wait until RWD data has been mounted before starting mmw-rwd upstart service

### PR DESCRIPTION
**Pull request by tnation14**
## Overview

Configure the `mmw-rwd` upstart service to wait for an `rwd-ready` event before starting, and configure cloud-init to emit an `rwd-ready` signal after the data volume is mounted. This will prevent the RWD service from starting before the data is availble. 

Connects #2357

### Demo

Optional. Screenshots, `curl` examples, etc.

### Notes

In order to guarantee that an upstart event wouldn't be emitted until after /opt/rwd-data was mounted, I removed the `mounts` cloud-init configuration and added a `mount` command to the `runcmd` list. I made sure the command to emit the upstart event runs after the `mount` command. 

## Testing Instructions
  * Configure the Upstart service to listen for an `rwd-ready` event in development
```diff
diff --git a/deployment/ansible/roles/model-my-watershed.rwd/templates/upstart-mmw-rwd.conf.j2 b/deployment/ansible/roles/model-my-watershed.rwd/templates/upstart-mmw-rwd.conf.j2
index 81e69cf9..f8c8fefe 100644
--- a/deployment/ansible/roles/model-my-watershed.rwd/templates/upstart-mmw-rwd.conf.j2
+++ b/deployment/ansible/roles/model-my-watershed.rwd/templates/upstart-mmw-rwd.conf.j2
@@ -1,7 +1,7 @@
 description "Rapid Watershed Delineation"
 
 {% if ['development', 'test'] | some_are_in(group_names) -%}
-start on (vagrant-mounted and started docker)
+start on (vagrant-mounted and started docker and rwd-ready)
 {% else %}
 start on (filesystem and local-filesystems and started docker and rwd-ready)
 {% endif %}
(END)
```

  * Provision the worker VM
  * Restart the worker VM, and confirm that the RWD service isn't running
```bash
$ vagrant provision worker
$ vagrant reload worker
$ vagrant ssh worker -c "sudo service mmw-rwd status"
mmw-rwd stop/waiting
```

  * Modify the worker's shell provisioner to emit `rwd-ready` event, and run the provisioner. Verify that the RWD service is running.
```diff
diff --git a/Vagrantfile b/Vagrantfile
index e604ad6d..ec6731dc 100644
--- a/Vagrantfile
+++ b/Vagrantfile
@@ -119,7 +119,6 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
 
     worker.vm.provision "shell" do |shell|
       shell.inline = <<EOF
+      initctl emit rwd-ready
       service celeryd restart >> /dev/null 2>&1
 EOF
     end
```
```bash
$ vagrant provision --provision-with=shell worker
$ vagrant ssh worker -c "sudo service mmw-rwd status"
mmw-rwd start/running, process 2423
```

* I also successfully tested this change on staging. I deployed once without updating the `cloud-config` to emit an upstart event; when I skipped this step `mmw-rwd` never started on the workers. The second time I deployed after modifying cloud-config to emit the upstart event, and  `mmw-rwd` started successfully.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -117,7 +117,11 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
       ansible.raw_arguments = ["--timeout=60"]
     end
 
-    worker.vm.provision "shell", inline: "service celeryd restart >> /dev/null 2>&1", run: "always"
+    worker.vm.provision "shell" do |shell|
```
Comment (hectcastro) [inline Vagrantfile]: I believe that this is now missing the `run: "always"` option. Not entirely sure this provisioner is still necessary, but I think it would be safest to keep things as they were since this change doesn't seem related to the PR.

File: deployment/cfn/worker.py [hunk 2]
```diff
@@ -420,6 +417,8 @@ def get_cloud_config(self):
                 '    content: ', self.get_input('RollbarServerSideAccessToken'),  # NOQA
                 '\n',
                 'runcmd:\n',
+                '  - mount -o "defaults,nofail,discard" -t ext4 /dev/xvdf /opt/rwd-data\n',
```
Comment (hectcastro) [inline deployment/cfn/worker.py]: I believe that the `cloud-config` `mount` directive also took care of persisting the mount in `/etc/fstab`. I checked out the instances on staging and did not see any entires in `fstab`, so I think we may need to replicate that functionality here.

Comment (tnation14) [reply deployment/cfn/worker.py]: In 7e44ff1, I modified cloud-config to create `fstab.rwd-data`, which contains an `fstab` entry for `/dev/xvdf`.  Then I added a `runcmd` entry to add the contents of `fstab.rwd-data` to `/etc/fstab`. I figured doing it that way would make it easier to make changes and add new volumes in the future.

**Comment by tnation14** [comment]
This is ready for another look.
